privacy/gdpr/consentwriter.go

```diff
@@ -19,21 +19,18 @@ func (c ConsentWriter) Write(req *openrtb2.BidRequest) error {
 	reqWrap := &openrtb_ext.RequestWrapper{BidRequest: req}
 
 	if c.RegExtGDPR != nil {
-		if regsExt, err := reqWrap.GetRegExt(); err == nil {
-			regsExt.SetGDPR(c.RegExtGDPR)
-		} else {
-			return err
+		if reqWrap.Regs != nil {
+			reqWrap.Regs.GDPR = c.RegExtGDPR
 		}
 	}
 
 	if c.Consent != "" {
-		if userExt, err := reqWrap.GetUserExt(); err == nil {
-			userExt.SetConsent(&c.Consent)
-		} else {
-			return err
+		if reqWrap.User != nil {
+			reqWrap.User.Consent = c.Consent
 		}
 	}
 
+	// do we need to rebuild req here?
```

Comment:
If we proceed with making your proposed changes such that all references to exts are deleted, then we don't need to rebuild the request here.